phpBB/phpbb/session.php

```diff
@@ -964,16 +964,17 @@ function session_gc()
 		}
 
 		/**
-		 * Get most recent session for each registered user to sync user last visit with it
-		 * Inner SELECT gets most recent sessions for each unique session_user_id
+		 * Get expired sessions for registered users, only most recent for each user
+		 * Inner SELECT gets most recent expired sessions for unique session_user_id
 		 * Outer SELECT gets data for them
 		 */
 		$sql_select = 'SELECT s1.session_page, s1.session_user_id, s1.session_time AS recent_time
 			FROM ' . SESSIONS_TABLE . ' AS s1
 			INNER JOIN (
 				SELECT session_user_id, MAX(session_time) AS recent_time
 				FROM ' . SESSIONS_TABLE . '
-				WHERE session_user_id <> ' . ANONYMOUS . '
+				WHERE session_time < ' . ($this->time_now - (int) $config['session_length']) . '
```

Comment:
This will select only expired sessions which is not what we need I think.

Reply:
And possible proposal. For the index page, `session_last_visit` is used to display last visit time. It could be used for profiles possibly then.
https://github.com/phpbb/phpbb/blob/4038bb9dc3a49197046fe3ef0f1b807866250d3f/phpBB/includes/functions.php#L3856
![image](https://github.com/user-attachments/assets/8bc33328-a1a6-440a-9417-cf23e48ee8f3)

Also, in this case we would get rid of all those huge SQL queries at all as they're only used to update user last visit.

Reply:
This change essentially reverts the last visit behavior back to what it was before 3.3.12. It has worked fine before and used the expired sessions as a way of determining the last visit time.